OpenRA.Mods.Common/Widgets/Logic/ReplayBrowserLogic.cs

```diff
@@ -66,31 +69,18 @@ public ReplayBrowserLogic(Widget widget, ModData modData, Action onExit, Action
 				ThreadPool.QueueUserWorkItem(_ => LoadReplays(dir, template));
 
 			var watch = panel.Get<ButtonWidget>("WATCH_BUTTON");
-			watch.IsDisabled = () => selectedReplay == null || selectedReplay.GameInfo.MapPreview.Status != MapStatus.Available;
+			watch.IsDisabled = () => selectedReplay == null || Map.Status != MapStatus.Available;
 			watch.OnClick = () => { WatchReplay(); };
 
 			panel.Get("REPLAY_INFO").IsVisible = () => selectedReplay != null;
 
-			var preview = panel.Get<MapPreviewWidget>("MAP_PREVIEW");
-			preview.SpawnOccupants = () => selectedSpawns;
-			preview.Preview = () => selectedReplay != null ? selectedReplay.GameInfo.MapPreview : null;
-
-			var titleLabel = panel.GetOrNull<LabelWidget>("MAP_TITLE");
-			if (titleLabel != null)
+			Ui.LoadWidget("MAP_PREVIEW", panel.Get("MAP_PREVIEW_ROOT"), new WidgetArgs
 			{
-				titleLabel.IsVisible = () => selectedReplay != null;
-
-				var font = Game.Renderer.Fonts[titleLabel.Font];
-				var title = new CachedTransform<MapPreview, string>(m => WidgetUtils.TruncateText(m.Title, titleLabel.Bounds.Width, font));
-				titleLabel.GetText = () => title.Update(selectedReplay.GameInfo.MapPreview);
-			}
-
-			var type = panel.GetOrNull<LabelWidget>("MAP_TYPE");
-			if (type != null)
-			{
-				var mapType = new CachedTransform<MapPreview, string>(m => m.Categories.FirstOrDefault() ?? "");
-				type.GetText = () => mapType.Update(selectedReplay.GameInfo.MapPreview);
-			}
+				{ "orderManager", null },
+				{ "getMap", (Func<MapPreview>)(() => Map) },
+				{ "onMouseDown",  (Action<MapPreviewWidget, MapPreview, MouseInput>)((preview, map, mi) => { }) },
+				{ "getSpawnOccupants", (Func<MapPreview, Dictionary<CPos, SpawnOccupant>>)(map => LobbyUtils.GetSpawnOccupants(selectedReplay.GameInfo.Players, map)) },
+			});
 
 			panel.Get<LabelWidget>("DURATION").GetText = () => WidgetUtils.FormatTimeSeconds((int)selectedReplay.GameInfo.Duration.TotalSeconds);
```

Comment:
Can you please change this to the following while you're here?
```csharp
var replayDuration = new CachedTransform<ReplayMetadata, string>(r =>
	"Duration: {0}".F(WidgetUtils.FormatTimeSeconds((int)selectedReplay.GameInfo.Duration.TotalSeconds)));
panel.Get<LabelWidget>("DURATION").GetText = () => replayDuration.Update(selectedReplay);
```
This makes the label look less disconnected when the install map button is showing, plus a small perf win.